admin/tabs/general/alertBanner/AlertBannerModel.ts

```diff
@@ -67,6 +73,13 @@ export class AlertBannerModel extends HoistModel {
         ];
     }
 
+    get clientAppOptions() {
+        return XH.getConf('xhClientApps', []).map(app => ({
```

Comment:
Don and I discussed the pros and cons of this approach - an alternative might involve having dev-utils inject a list, with some tradeoffs between the two options.  Let us discuss!

Reply:
or have it specified in js code in the admin app model?

Reply:
Anselm and I decided in the end to go the dev-utils injection approach. We thought this approach had the least annoying issues compared to the other two where it would force users to update dev-utils if they want the latest hoist-react and not get display names to populate dropdowns, using the app codes instead. It generally requires the least amount of additional effort in that if the clients have apps set up with clientAppCodes prior, then the select field would just automatically populate. This does assume the app codes are reasonably differentiating, like "contacts" vs "app12".

With the config approach, we run into the issue of user error being a bit punishing for when an admin misspells codes in spite of entering in the correct JSON format within the production app itself where a banner that was expected to appear just doesn't. Also, if clients want consistency in behavior between different stages, then the config would have to be copied at each level every time a change would be made (though this is infrequent).

Lastly with the app model approach, we thought that not all Hoist consumers would extend the exported AppModel off Admin, opting to just use the Admin as is given it's not an abstract class. By making it hardcoded, we'd force them to create a class that extends that if they already haven't, and it just seems like a clunky way to add in app codes, especially if the extension is only really just for the codes.

Reply:
👍🏻